From d7d3d8daa6cbd6ad5c67a9184c936341b213f860 Mon Sep 17 00:00:00 2001
From: "Chen, Yu" <yu.y.chen@intel.com>
Date: Tue, 1 Mar 2022 16:48:59 +0800
Subject: [PATCH 13/21] Add 32-bit RGBX/RGBA-SRGB formatsfix black screen

issue of game xsmdl

Add PIPE_FORMAT_RGBA8888_SRGB and PIPE_FORMAT_RGBX8888_SRGB in dri2.c and dri_screen.c,
to support android game to call eglCreateWindowSurface with colorspace value set as EGL_GL_COLORSPACE_SRGB_KHR

Signed-off-by: Chen, Yu <yu.y.chen@intel.com>
Change-Id: I0dcdc9b73637f8a221b93b395d488056ad260a5d
Signed-off-by: Marc Mao <marc.mao@intel.com>
---
 src/gallium/frontends/dri/dri2.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gallium/frontends/dri/dri2.c b/src/gallium/frontends/dri/dri2.c
index df504124285..099d9d21f94 100644
--- a/src/gallium/frontends/dri/dri2.c
+++ b/src/gallium/frontends/dri/dri2.c
@@ -179,6 +179,7 @@ dri2_drawable_get_buffers(struct dri_drawable *drawable,
       case PIPE_FORMAT_R10G10B10A2_UNORM:
       case PIPE_FORMAT_BGRA8888_UNORM:
       case PIPE_FORMAT_RGBA8888_UNORM:
+      case PIPE_FORMAT_RGBA8888_SRGB:
 	 depth = 32;
 	 break;
       case PIPE_FORMAT_R10G10B10X2_UNORM:
@@ -187,6 +188,7 @@ dri2_drawable_get_buffers(struct dri_drawable *drawable,
          break;
       case PIPE_FORMAT_BGRX8888_UNORM:
       case PIPE_FORMAT_RGBX8888_UNORM:
+      case PIPE_FORMAT_RGBX8888_SRGB:
 	 depth = 24;
 	 break;
       case PIPE_FORMAT_B5G6R5_UNORM:
@@ -274,9 +276,11 @@ dri_image_drawable_get_buffers(struct dri_drawable *drawable,
       case PIPE_FORMAT_BGRA8888_UNORM:
          image_format = __DRI_IMAGE_FORMAT_ARGB8888;
          break;
+      case PIPE_FORMAT_RGBX8888_SRGB:
       case PIPE_FORMAT_RGBX8888_UNORM:
          image_format = __DRI_IMAGE_FORMAT_XBGR8888;
          break;
+      case PIPE_FORMAT_RGBA8888_SRGB:
       case PIPE_FORMAT_RGBA8888_UNORM:
          image_format = __DRI_IMAGE_FORMAT_ABGR8888;
          break;
-- 
2.25.1

